addons/product/models/product.py

```diff
@@ -218,10 +218,15 @@ def _compute_is_product_variant(self):
         for product in self:
             product.is_product_variant = True
 
-    @api.depends_context('pricelist', 'partner', 'quantity', 'uom', 'date', 'no_variant_attributes_price_extra')
+    @api.depends_context('tuple(pricelist or [])', 'partner', 'quantity', 'uom', 'date', 'no_variant_attributes_price_extra')
     def _compute_product_price(self):
         prices = {}
         pricelist_id_or_name = self._context.get('pricelist')
+        pricelist_id_or_name = (
+            pricelist_id_or_name[0]
+            if isinstance(pricelist_id_or_name, list) and pricelist_id_or_name
+            else pricelist_id_or_name
+        )
```

Comment:
Might be clearer with a regular conditional e.g.
```python
if pricelist_id_or_name and isinstance(pricelist_id_or_name, list):
    pricelist_id_or_name = pricelist_id_or_name[0]
```
In fact that could even be moved a few lines down as line 230 is already checking `if pricelist_id_or_name` so it could just have one more conditional for cleaning up the sequence just before the existing type dispatch.